GeneratorInterface/RivetInterface/plugins/GenParticles2HepMCConverter.cc

```diff
@@ -210,7 +210,7 @@ void GenParticles2HepMCConverter::produce(edm::Event& event, const edm::EventSet
 
   // Finalize HepMC event record
   std::unique_ptr<edm::HepMC3Product> hepmc_product(new edm::HepMC3Product());
-  hepmc_product->addHepMCData(hepmc_event);
```

Comment:
Could you add a comment here stating that HepMC3Product does not take ownership of the pointer? Else it looks like this is also a memory problem as hepmc_event gets deleted at the end of this routine.

Alternatively, you could change hepmc_event into a stack variable, i.e. `HepMCe::GenEvent hepmc_event;` and then it is clear that the lifetime is only for the function.

Reply:
Sounds good -- I changed it to a stack variable.